sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java

```diff
@@ -212,7 +193,12 @@ public OfferEvaluator(
             evaluationPipeline.add(new ExecutorEvaluationStage(Optional.empty()));
             evaluationPipeline.addAll(getNewEvaluationPipeline(podInstanceRequirement, allTasks, tlsStageBuilder));
         } else {
-            evaluationPipeline.add(new ExecutorEvaluationStage(getExecutorInfo(thisPodTasks.values())));
+            Optional<Protos.ExecutorInfo> executorInfo = getExecutorInfo(thisPodTasks.values());
+            String executorIdString = executorInfo.get().getExecutorId().getValue();
```

Comment:
Is it actually set to an empty string sometimes? If just checking for presence can do `executorInfo.get().hasExecutorId()`. String fields which are unset will return `""` if you `getX()` against them anyway.